x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_management/components/analytics_list/actions.tsx

```diff
@@ -29,87 +29,123 @@ import { stopAnalytics } from '../../services/analytics_service';
 import { StartAction } from './action_start';
 import { DeleteAction } from './action_delete';
 
-export const AnalyticsViewAction = {
-  isPrimary: true,
-  render: (item: DataFrameAnalyticsListRow) => {
-    const analysisType = getAnalysisType(item.config.analysis);
-    const isDisabled =
-      !isRegressionAnalysis(item.config.analysis) &&
-      !isOutlierAnalysis(item.config.analysis) &&
-      !isClassificationAnalysis(item.config.analysis);
-
-    const url = getResultsUrl(item.id, analysisType);
-    return (
-      <EuiButtonEmpty
-        isDisabled={isDisabled}
-        onClick={() => (window.location.href = url)}
-        size="xs"
-        color="text"
-        iconType="visTable"
-        aria-label={i18n.translate('xpack.ml.dataframe.analyticsList.viewAriaLabel', {
-          defaultMessage: 'View',
-        })}
-        data-test-subj="mlAnalyticsJobViewButton"
-      >
-        {i18n.translate('xpack.ml.dataframe.analyticsList.viewActionName', {
-          defaultMessage: 'View',
-        })}
-      </EuiButtonEmpty>
-    );
-  },
+interface Props {
+  item: DataFrameAnalyticsListRow;
+  isManagementTable: boolean;
+}
+
+const AnalyticsViewButton: FC<Props> = ({ item, isManagementTable }) => {
+  const {
+    services: {
+      application: { navigateToUrl, navigateToApp },
+    },
+  } = useMlKibana();
+
+  const analysisType = getAnalysisType(item.config.analysis);
+  const isDisabled =
+    !isRegressionAnalysis(item.config.analysis) &&
+    !isOutlierAnalysis(item.config.analysis) &&
+    !isClassificationAnalysis(item.config.analysis);
+
+  const url = getResultsUrl(item.id, analysisType);
+  const navigator = isManagementTable
+    ? () => navigateToApp('ml', { path: url })
+    : () => navigateToUrl(url);
+
+  return (
+    <EuiButtonEmpty
+      isDisabled={isDisabled}
+      onClick={navigator}
+      size="xs"
+      color="text"
+      iconType="visTable"
+      aria-label={i18n.translate('xpack.ml.dataframe.analyticsList.viewAriaLabel', {
+        defaultMessage: 'View',
+      })}
+      data-test-subj="mlAnalyticsJobViewButton"
+    >
+      {i18n.translate('xpack.ml.dataframe.analyticsList.viewActionName', {
+        defaultMessage: 'View',
+      })}
+    </EuiButtonEmpty>
```

Comment:
question: could it be just a link?

Reply:
In this case it needs to work from the management page and the list so it feels clearer to keep it as is so we can cover both cases with straight-forward logic.